sdk/storage/storage-file-share/test/fileserviceclient.spec.ts

```diff
@@ -380,4 +385,83 @@ describe("FileServiceClient", () => {
     assert.ok(typeof result.requestId);
     assert.ok(result.requestId!.length > 0);
   });
+
+  it("ListShares with deleted share", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+      }
+    }
+    assert.ok(found);
+  });
+
+  it("Undelete share positive", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    let shareDeleted: ShareItem | undefined;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+        assert.ok(share.properties.deletedTime);
+        assert.ok(share.properties.remainingRetentionDays);
+
+        shareDeleted = share;
+      }
+    }
+    assert.ok(found);
+    assert.ok(shareDeleted);
+
+    // Await share to be deleted.
+    await delay(30 * 1000);
+
+    const restoredShareClient = await serviceClient.undeleteShare(
+      shareDeleted!.name,
+      shareDeleted!.version!
+    );
+    await restoredShareClient.getProperties();
+
+    await restoredShareClient.delete();
+  });
+
+  it("Undelete share negative", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
```

Comment:
I've made a new issue with feedback I have regarding the test folders and these skips here: https://github.com/Azure/azure-sdk-for-js/issues/10796

I don't think we should address that issue as part of this PR! But I wanted to bring your attention to the issue I made, so that we could continue talking about these ideas there.

Reply:
Thanks for the info!

---
In reply to: [475542800](https://github.com/Azure/azure-sdk-for-js/pull/10792#discussion_r475542800) [](ancestors = 475542800)